packages/lesswrong/server/fmCrosspost/crosspost.ts

```diff
@@ -0,0 +1,91 @@
+import fetch from "node-fetch";
+import Users from "../../lib/collections/users/collection";
+import { randomId } from "../../lib/random";
+import { Crosspost, UpdateCrosspostPayload, CrosspostPayload } from "./types";
+import { signToken } from "./tokens";
+import { apiRoutes, makeApiUrl } from "./routes";
+import { makeCrossSiteRequest } from "./resolvers";
+
+export const performCrosspost = async <T extends Crosspost>(post: T): Promise<T> => {
+  if (!post.fmCrosspost || !post.userId || post.draft) {
+    return post;
+  }
+
+  const {isCrosspost, hostedHere, foreignPostId} = post.fmCrosspost;
+  if (!isCrosspost || !hostedHere || foreignPostId) {
+    return post;
+  }
+
+  const user = await Users.findOne({_id: post.userId});
+  if (!user || !user.fmCrosspostUserId) {
+    throw new Error("You have not connected a crossposting account yet");
+  }
+
+  const token = await signToken<CrosspostPayload>({
+    localUserId: post.userId,
+    foreignUserId: user.fmCrosspostUserId,
+  });
+
+  // If we're creating a new post without making a draft first then we won't have an ID yet
+  if (!post._id) {
+    post._id = randomId();
+  }
+
+  const apiUrl = makeApiUrl(apiRoutes.crosspost);
+  const result = await fetch(apiUrl, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      token,
+      postId: post._id,
+      postTitle: post.title,
+    }),
+  });
+  const json = await result.json();
+  if (json.status !== "posted" || !json.postId) {
+    throw new Error(`Failed to create crosspost: ${JSON.stringify(json)}`);
```

Comment:
Is it possible to surface more context here? Unclear to me how to fix this:

`Failed to create crosspost: {"error":"app.validation_error"}`

Tried unlinking + relinking accounts and removing tags not present on both sites to no avail

Reply:
Thanks for reporting the issue - this should now be fixed!